src/windows.rs

```diff
@@ -117,6 +131,20 @@ impl MmapInner {
         }
     }
 
+    pub fn set_protection(&mut self, prot: Protection) -> io::Result<()> {
+        let mut old = 0;
+        let result = unsafe { kernel32::VirtualProtect(self.ptr,
```

Comment:
I think this needs to take into account the offset into the page of `self.ptr` for when this is a file-backed mmap specified with an offset (see `drop` for an example).